monitoring/jmx-grafana-prometheus-step-by-step/prometheus-config-map.yml

```diff
@@ -0,0 +1,186 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: prometheus-server-conf
+  labels:
+    name: prometheus-server-conf
+  namespace: monitoring
+data:
+  prometheus.rules: |-
+    groups:
+    - name: example
+      rules:
+
+      # Alert for any instance that is unreachable for >2 minutes.
+      - alert: service_down
+        expr: up == 0
+        for: 2m
+        labels:
+          severity: page
+        annotations:
+          summary: "Instance {{ $labels.instance }} down"
+          description: "{{ $labels.instance }} of job {{ $labels.job }} has been down for more than 2 minutes."
+
+      - alert: high_load
+        expr: node_load1 > 0.5
+        for: 2m
+        labels:
+          severity: page
+        annotations:
+          summary: "Instance {{ $labels.instance }} under high load"
+          description: "{{ $labels.instance }} of job {{ $labels.job }} is under high load."
+
+  prometheus.yml: |-
+    global:
+      # How frequently to scrape targets by default
+      # Default 15s
+      scrape_interval: 30s
+      # How frequently to evaluate rules.
+      # Default 15s
+      evaluation_interval: 15s
+      # How long until a scrape request times out.
+      # Default to 10s.
+      # Required because cp-demo is using cpu throttling, so let's leave enough time to fetch the metrics in particular for the first time as it needs to compile all rexps
+      scrape_timeout: 30s
+
+    rule_files:
+      - "alert.rules"
+
+    alerting:
+      alertmanagers:
+        - scheme: http
+          static_configs:
+            - targets:
+                - "alertmanager:9093"
+
+    scrape_configs:
+      - job_name: "prometheus"
+        static_configs:
+          - targets: ["localhost:9090"]
+          
+      - job_name: "zookeeper"
+        static_configs:
+          - targets:
+              - "zookeeper-0.zookeeper.confluent.svc.cluster.local:7778"
+              - "zookeeper-1.zookeeper.confluent.svc.cluster.local:7778"
+              - "zookeeper-2.zookeeper.confluent.svc.cluster.local:7778"
```

Comment:
Could you use a ServiceMonitor from Prometheus Operator / kube-prometheus to simplify this?

Reply:
@tsuz Could you take a look at the request above?

Reply:
@OneCricketeer @arodoni 
Apologizes. I have not been able to take a look at these yet.